gitspy/Parseable.swift

```diff
@@ -9,7 +9,31 @@
 import Foundation
 
 protocol Parseable {
-    typealias A
-    func parse(data: AnyObject) -> A?
+    
+    init?(dictionary: [String: AnyObject])
+    init?(data: AnyObject)
+    
+    // there could be a cooler way with Array extension + generics...
+    static func array(objects: [[String: AnyObject]]) -> [Self]
 }
 
+extension Parseable {
+    
+    init?(data: AnyObject) {
+        guard let dictionary = data as? [String: AnyObject] else {
+            return nil
+        }
+        self.init(dictionary: dictionary)
+    }
+
+    static func array(objects: [[String: AnyObject]]) -> [Self] {
+        var array = [Self]()
+        for dictionary in objects {
+            if let object = Self(dictionary: dictionary) {
+                array.append(object)
+            }
+        }
+        return array
+    }
```

Comment:
I'm also having problems naming stuff. Because in swift you should have description like `arrayWithObjects` but array alone sounds weird.
Today I wanted : indexForExercise but this is a wrong name in swift, however index(exercise) is not enough for me